src/windows/tvgWindowImpl.h

```diff
@@ -0,0 +1,246 @@
+        glsl_version = "#version 150";
+        glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+        glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 2);
+        glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);  // 3.2+ only
+        glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);            // Required on Mac
+    #else
+        // GL 3.0 + GLSL 130
+        glsl_version = "#version 130";
+        glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+        glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
+        //glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);  // 3.2+ only
+        //glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);            // 3.0+ only
+    #endif
+
+        gl_window = glfwCreateWindow(w, h, name.c_str(), NULL, NULL);
+        if (!gl_window) {
+            glfwTerminate();
+            return;
+        }
+
+        glfwSetFramebufferSizeCallback(gl_window, Window::Impl::glfwOnFramebufferResize);
+
+        glfwMakeContextCurrent(gl_window);
+        glfwSwapInterval(1);
+        glEnable(GL_TEXTURE_2D);
+
+        tvg::Initializer::init(tvg::CanvasEngine::Sw, std::thread::hardware_concurrency());
+        canvas = tvg::SwCanvas::gen();
+
+        glfwGetWindowSize(gl_window, &width, &height);
+        resize(width, height);
+
+        lastTime = glfwGetTime();
+
+        printf("Called WindowImpl Constructor\n");
+    }
+
+    ~Impl()
+    {
+        printf("Called WindowImpl Destructor\n");
+    }
+
+    void close()
+    {
+
+        printf("Called WindowImpl close()\n");
+    }
+
+    bool run()
+    {
+        printf("Called WindowImpl run()\n");
+
+        // Exit if there was an error during setup
+        if (!instance_window) return false;
+
+
+        // User setup
+        //setup();
+
+        // Main render loop
+        while (!glfwWindowShouldClose(gl_window))
+        {
+            // Clear
+            glClearColor(((clearColor >> 16) & 0xff) / 255.0f, ((clearColor >> 8) & 0xff) / 255.0f, (clearColor & 0xff) / 255.0f, ((clearColor >> 24) & 0xff) / 255.0f);
+            glClear(GL_COLOR_BUFFER_BIT);
+
+            // User render loop
+            double thisTime = glfwGetTime();
+            fps = 1.0 / (thisTime - lastTime);
+
+
+            auto main_scene = tvg::Scene::gen();
+            bool updated = on_update(main_scene.get());
+            canvas->push(std::move(main_scene));
+            //
+
+            //update(thisTime - lastTime);
+            if (updated && canvas->draw() == tvg::Result::Success) canvas->sync();
+
+            // Render the buffer to a texture and display it
+            glBindTexture(GL_TEXTURE_2D, texture);
+            glTexSubImage2D(
+                GL_TEXTURE_2D,
+                0,
+                0, 0,
+                width, height,
+                GL_RGBA,
+                GL_UNSIGNED_BYTE,
+                (void*)buffer
+            );
+            glBegin(GL_QUADS); 
+            glTexCoord2f(0, 0); glVertex2f(0, 0);
```

Comment:
If the texture is meant to cover the hole window, I think it's more convenient to use `glBlitFramebuffer`